pkgs/by-name/mu/musl/package.nix

```
@@ -0,0 +1,177 @@
+{
+  lib,
+  callPackage,
+  fetchurl,
+  linuxHeaders,
+  stdenv,
+  # Boolean flags
+  hasLinuxHeaders ? true,
+  useBSDCompatHeaders ? true,
+}:
+let
+  _sources = callPackage ./sources.nix { };
+  arch =
+    if stdenv.hostPlatform.isx86_64 then
+      "x86_64"
+    else if stdenv.hostPlatform.isx86_32 then
+      "i386"
+    else null;
+in
+stdenv.mkDerivation (finalAttrs: {
+  inherit (_sources.musl) pname version src;
+
+  env = {
+    NIX_DONT_SET_RPATH = true;
+    CFLAGS = toString ([
+      "-fstack-protector-strong"
+    ] ++ lib.optionals stdenv.hostPlatform.isPower [
+      "-mlong-double-64"
+    ]);
+  };
+
+  configureFlags = [
+    (lib.enableFeature true "shared")
+    (lib.enableFeature true "static")
+    (lib.enableFeature true "debug")
+    (lib.enableFeatureAs true "wrapper" "all")
+    "--syslibdir=${placeholder "out"}/lib"
+  ];
+
+  outputs = [
+    "out"
+    "bin"
+    "dev"
+  ];
+
+  outputLib = "out";
+
+  # Disable auto-adding stack protector flags, so musl can selectively disable
+  # as needed
+  hardeningDisable = [ "stackprotector" ];
+
+  strictDeps = true;
+
+  enableParallelBuilding = true;
+
+  dontDisableStatic = true;
+
+  dontAddStaticConfigureFlags = true;
+
+  separateDebugInfo = true;
+
+  # Let's be friendlier to debuggers/perf tools
+  # Neither force them on, nor force them off
+  postPatch = ''
+    substituteInPlace configure \
+      --replace -fno-unwind-tables "" \
+      --replace -fno-asynchronous-unwind-tables ""
+  '';
+
+  # the `-x c` flag is required here because the file extension confuses gcc,
+  # that will regard the file as a linker script.
+  preBuild = lib.optionalString (stdenv.targetPlatform.libc == "musl" && stdenv.targetPlatform.isx86_32) ''
+    $CC -x c -c ${_sources.stack_chk_fail_local_c.src} -o __stack_chk_fail_local.o
+    $AR r libssp_nonshared.a __stack_chk_fail_local.o"
+  '';
+
+  postInstall = lib.concatStringsSep "\n" [
+    # Not sure why, but link in all but scsi directory as that's what
+    # uclibc/glibc do. Apparently glibc provides scsi itself?
+    (lib.optionalString hasLinuxHeaders
+      ''
+        pushd ''${!outputDev}/include
+        ln -s $(ls -d ${linuxHeaders}/include/* | grep -v "scsi$") .
+        popd
+      '')
+
+    (lib.optionalString (stdenv.targetPlatform.libc == "musl" && stdenv.targetPlatform.isx86_32)
+      ''install -D libssp_nonshared.a ''${!outputLib}/lib/libssp_nonshared.a'')
+
+    # Create 'ldd' symlink, builtin
+    ''ln -s ''${!outputLib}/lib/libc.so ''${!outputBin}/bin/ldd''
+
+    # (impure) cc wrapper around musl for interactive usuage
+    ''
+      for i in musl-gcc musl-clang ld.musl-clang; do
+        moveToOutput bin/$i ''${!outputDev}
+      done
+      moveToOutput lib/musl-gcc.specs ''${!outputDev}
+      substituteInPlace ''${!outputDev}/bin/musl-gcc \
+        --replace ''${!outputLib}/lib/musl-gcc.specs ''${!outputDev}/lib/musl-gcc.specs
```

```suggestion
        --replace ''${!outputLib}/lib/musl-gcc.specs ''${!outputDev}/lib/musl-gcc.specs
```
`replace` is deprecated